spec.md

```diff
@@ -840,6 +841,21 @@ If the image had already been deleted or did not exist, a `404 Not Found` respon
 
 > for more details, see: [compatibility.md](https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-2.md#backward-compatibility)
 
+### Registry Proxying
+
+A registry MAY operate as a proxy to another registry to support pull-through caching or proxying of pull operations (such as fetching tags, manifests, or blobs).
+In this case, the `Host` header passed to the proxy registry WILL NOT be the same as the host in the repository name used by the client.
+A `ns` query parameter on pull operations is OPTIONAL, but when used specifies the host in a repository name used by a client.
+This host component in a repository name SHOULD be the registry host a client considers the primary source for a repository, however, a client MAY be configured to use a different host.
+The host in the repository name SHOULD be the first component of the full repository name used by a client.
+This original host component used by the client is referred to as the source host in the API documentation.
+A proxy registry MAY use the `ns` query parameter to resolve an upstream registry host.
+A proxy registry MAY choose to ignore the `ns` query parameter.
+
+A client SHOULD be aware of whether a registry host is being used is a proxy.
```

Comment:
This leads to the question of "how do I do that?"

Reply:
http headers that the client SHOULD log somewhere?

Reply:
Sure, the case is usually when http `HOST` does not match the source host in the `ns` query parameter. The client can obviously define their own rules, but can mention this condition as a recommendation based on parts within this specification.